metadataserver/inject/inject_linux.go

```diff
@@ -338,3 +364,59 @@ done:
 
 	return &socket, nil
 }
+
+func getLinkLocalAddr(ctx context.Context, handle *netlink.Handle, link netlink.Link) (*netlink.Addr, error) {
+	t := time.NewTimer(1)
+	defer t.Stop()
+	for {
+		select {
+		case <-ctx.Done():
+			return nil, fmt.Errorf("Context complete before able to get LL addrs: %w", ctx.Err())
+		case <-t.C:
+			addrs, err := handle.AddrList(link, netlink.FAMILY_V6)
+			if err != nil {
+				return nil, fmt.Errorf("Could not list V6 addrs on %q: %w", link.Attrs().Name, err)
+			}
+			for idx := range addrs {
+				addr := addrs[idx]
+				logger.G(ctx).Debugf("Checking addr %v, with flags %d", addr.IP, addr.Flags)
+				if addr.Scope == int(netlink.SCOPE_LINK) {
+					return &addr, nil
+				}
+			}
+		}
+		t.Reset(10 * time.Millisecond)
+	}
+}
+
+func waitForInterfacesUp(ctx context.Context, containerNSHandle, intermediateNSHandle *netlink.Handle) error {
+	t := time.NewTimer(1)
+	defer t.Stop()
+	var err error
+	var toIMDSLink, toContainerLink netlink.Link
+	for {
+		select {
+		case <-ctx.Done():
+			return fmt.Errorf("Context complete before expected: %w, last states; toIMDSLink: %v, toContainerLink:%v", ctx.Err(), toIMDSLink, toContainerLink)
+		case <-t.C:
+			toIMDSLink, err = containerNSHandle.LinkByName(toimdsInterfaceName)
+			if err != nil {
+				return fmt.Errorf("Could not get %s: %w", toimdsInterfaceName, err)
+			}
+
+			toContainerLink, err = intermediateNSHandle.LinkByName(tocontainerInterfaceName)
+			if err != nil {
+				return fmt.Errorf("Could not get %s: %w", tocontainerInterfaceName, err)
+			}
+
```

Comment:
Do we have to continuously poll to get these? 
By the time we run this wait function, is it not safe to assume the Links themselves exist?

Reply:
In order to get the current _status_ of the link, we have to refetch the data. Netlink also has a "monitor" API, but the golang library doesn't suppot it as far as I can tell.